site/content/integrate/plugins/components/server/best-practices.md

```diff
@@ -12,6 +12,12 @@ aliases:
 
 Add all static files under a file directory named `public` within the plugin directory, and include the files in the plugin bundle using the Makefile.
 
-## How can plugins make sure http requests are authentic?
+## How do plugins make sure http requests are authentic?
 
-Plugins can implement the [`ServeHTTP`]({{< ref "/integrate/reference/server/server-reference#Hooks.ServeHTTP" >}}) to listen to http requests. This can e.g. be used to receive post action requests when [Interactive Messages Buttons and Menus](https://docs.mattermost.com/developer/interactive-messages.html) are triggered by users. Since these requests are just http requests, anyone can send them to the plugin. Hence the plugin must make sure the requests are authentic. The Mattermost Server sets the http header `Mattermost-User-Id` if and only if the request is made by an authenticated client. The plugin therefore has to only check if this header is set and reject all other requests.
+Plugins can implement the [`ServeHTTP`]({{< ref "/integrate/reference/server/server-reference#Hooks.ServeHTTP" >}}) to listen to http requests. This can be used to receive post action requests when [Interactive Messages Buttons and Menus](https://docs.mattermost.com/developer/interactive-messages.html) are triggered by users.
+
+When plugins act as an HTTP server, they serve requests from Mattermost clients (which are authenticated in a Mattermost sense), but may also serve HTTP requests from external services like webhooks. These requests from external services might use the Authorization header to authorize themselves against the plugin.
+
+Since these requests are just HTTP requests, anyone can send them to the plugin. Hence the plugin must make sure the requests are authentic. The Mattermost Server sets the HTTP header ``Mattermost-User-Id`` when the request is made by an authenticated client. If the plugin is expecting the request to come from an authenticated Mattermost user, and the ``Mattermost-User-Id`` is blank, it's the plugin's responsibility to reject the request.
```

Comment:
```suggestion
Since these requests are just HTTP requests, anyone can send them to the plugin. Hence, the plugin must make sure the requests are authentic. The Mattermost Server sets the HTTP header ``Mattermost-User-Id`` when the request is made by an authenticated client. If the plugin is expecting the request to come from an authenticated Mattermost user, and the ``Mattermost-User-Id`` is blank, it's the plugin's responsibility to reject the request.
```